### PR TITLE
(#3567) - Reporting db.adapter in db.info

### PR DESCRIPTION
**Pull request by victorkohl**
This is my first pull request, so please let me know if I did anything wrong or something could be improved.

Implementing the reporting of db.adapter in db.info (#3567). Also reporting the backend *DOWN adapter.

### CODE REVIEW

File: lib/adapter.js [hunk 1]
```diff
@@ -700,6 +700,7 @@ AbstractPouchDB.prototype.info = utils.adapterFun('info', function (callback) {
     // assume we know better than the adapter, unless it informs us
     info.db_name = info.db_name || self._db_name;
     info.auto_compaction = !!(self.auto_compaction && self.type() !== 'http');
+    info.adapter = info.adapter || self.type();
```
Comment (daleharvey) [inline lib/adapter.js]: `type()` already has logic to fallback so info.adapter, so I think we should stick to 

    info.adapter = self.type();

(and scold ourself for calling it `.type()` in the first place

**Comment by daleharvey** [comment]
I cant remember exactly why we implemented `type()`, it looks very out of place but that is outside the scope of this PR, aside from the nit this looks great thanks :)

The tests are reporting a genuine failure
```
1) test.basics.js-http db.info should give adapter name (#3567):
      AssertionError: expected 'leveldb' to equal 'http'
      + expected - actual
      +"http"
      -"leveldb"
```

You should be able to reproroduce by doing  `SERVER=pouchdb-server npm test`

**Comment by nolanlawson** [comment]
+1 when green; this is helpful info to report!

It would also be good to document it [in this API page](https://github.com/pouchdb/pouchdb/blob/205f4a32585c5bbf05057c52b0a0bc526aba0f93/docs/_includes/api/database_information.html#L44-L48).

**Comment by victorkohl** [comment]
Thanks for the input guys, I made the suggested changes.

**Comment by daleharvey** [comment]
Sorry our tests arent 100% reliable right now, trying to fix that up but sometimes we need to restart them, im gonna do a quick test run to get saucelabs coverage on this them merge, it looks great thanks

**Comment by daleharvey** [comment]
Full green @ https://github.com/pouchdb/pouchdb/pull/3940

**Comment by daleharvey** [comment]
Awesome, thanks a lot @victorkohl, merged in https://github.com/pouchdb/pouchdb/commit/01b305791dfb58c86f3a66a8ba4c75e4a5f555c8

**Comment by nickcolley** [comment]
:tada: Thanks @victorkohl :)

**Pull request closed without merging**